docs/usage/strict_mode.md

```diff
@@ -0,0 +1,85 @@
+There are some situations where you want Pydantic to throw an error instead of coercing data.
```

Comment:
I don't think this file will fit well in the current organization. The section on the `models.md` about the Strict Mode that you added in this PR is great already.